src/main/java/bgu/spl/mics/application/passiveObjects/Squad.java

```diff
@@ -1,87 +1,129 @@
 package bgu.spl.mics.application.passiveObjects;
+
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
- * Passive data-object representing a information about an agent in MI6.
- * You must not alter any of the given public methods of this class. 
+ * Passive data-object representing a information about an agent in MI6. You must not alter any of the given public
+ * methods of this class.
  * <p>
  * You may add ONLY private fields and methods to this class.
+ *
  * @inv Squad is a singleton - only one instance of this class can be initialized
  */
 public class Squad {
-	private static Squad instance = null;
-	private Map<String, Agent> agents;
+    private static Squad _instance = null;
+    private Map<String, Agent> _agents;
 
-	/**
-	 * Retrieves the single instance of this class.
-	 * @pre none
-	 * @post return != null
-	 */
-	public static Squad getInstance() {
-		if (instance == null)
-			instance = new Squad();
-		return instance;
-	}
```

Comment:
Squad should be thread safe. Otherwise the multiple Moneypennys might cause problems.
some of this was explained [here](https://www.cs.bgu.ac.il/~spl201/index.php?page=PracticalSession08#thread_safe_singleton)